core/Access.php

```diff
@@ -138,19 +143,15 @@ public function __construct()
      */
     public function reloadAccess(Auth $auth = null)
     {
-        if (!is_null($auth)) {
-            $this->auth = $auth;
-        }
+        $this->resetSites();
 
-        // if the Auth wasn't set, we may be in the special case of setSuperUser(), otherwise we fail
-        if (is_null($this->auth)) {
-            if ($this->hasSuperUserAccess()) {
-                return $this->reloadAccessSuperUser();
-            }
+        if (isset($auth)) {
+            $this->auth = $auth;
         }
 
         if ($this->hasSuperUserAccess()) {
-            return $this->reloadAccessSuperUser();
```

Comment:
The user has already super user access, meaning we do not have to set `$this->hasSuperUserAccess = true`. We only need to make sure a login name is set